### PR TITLE
Fix setup and import errors from kaolin root directory

### PR DESCRIPTION
**Pull request by Jean-Francois-Lafleche**
### Description
- Follow Cython's recommended best practices by distributing the `.cpp` files and making Cython compilation optional (see https://cython.readthedocs.io/en/latest/src/userguide/source_files_and_compilation.html#distributing-cython-modules).
- Cython compilation can be triggered by setting the following env variable: `export USE_CYTHON=1` before running the setup.
- Cleanup `setup.py`, remove second setup script, add long description, add URL, improve readability.
- Update README with optional instruction to compile extensions inplace to prevent import errors when importing from the kaolin root directory.
- Update README to clarify requirements and installation.
- Update setup.py to pick up the version number indicated in `kaolin/version.py`.
- Update the project's version number to `0.1.0`.

### Related Issues
#40, #52, #43, #27, #21, #15, #14, #9, #3 

### What's Fixed
- Fix Cython related issues, Cython is no longer required during setup.
- Import errors when importing from kaolin root directory after `python setup.py install`.

### Tests
OS: Ubuntu 16.04
GPU: RTX 2080Ti
Driver: 435.21
CUDA: 10.1
Python: 3.6, 3.7

Commands:
- `python setup.py install; cd examples; python -c "import kaolin as kal; print(kal.__version__)"`
(note, this command by itself produces an import issue when importing kaolin from the kaolin root directory)
- `python setup.py build_ext --inplace; python setup.py install; python -c "import kaolin as kal; print(kal.__version__)"`
(note, when testing between different python versions, ensure to delete the `build` directory between tests)

### CODE REVIEW

File: setup2.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    'flake8',
+]
+
+
+# if __name__ == '__main__':
+# build_deps()
+setup(
+    # Metadata
+    name='f',
+    version='1',
```
Comment (krrish94) [inline setup2.py]: Should be `0.1.0`, to be consistent with everything else?

File: setup2.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        #     'kaolin/cuda/triangle_distance_cuda.cu',
+        # ]),
+        # CUDAExtension('kaolin.cuda.mesh_intersection', [
+        #     'kaolin/cuda/mesh_intersection.cpp',
+        #     'kaolin/cuda/mesh_intersection_cuda.cu',
+        # ]),
+        # CUDAExtension('kaolin.graphics.nmr.cuda.rasterize_cuda', [
+        #     'kaolin/graphics/nmr/cuda/rasterize_cuda.cpp',
+        #     'kaolin/graphics/nmr/cuda/rasterize_cuda_kernel.cu',
+        # ]),
```
Comment (krrish94) [inline setup2.py]: These could be safely removed (?)

File: setup2.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            'kaolin.nnsearch',
+            sources=[
+                'kaolin/cython/nnsearch.pyx'
+            ],
+            language='c++',
+        ),
+    ],
+    include_dirs=[np.get_include()],
+    # cmdclass={'build_ext': BuildExtension}
+)
```
Comment (krrish94) [inline setup2.py]: Need a newline here

File: setup.py [hunk 4]
```diff
@@ -10,7 +10,7 @@
 VERSION = '0.1.0'
 DESCRIPTION = 'Kaolin: A PyTorch library for accelerating 3D deep learning research'
 URL = 'https://github.com/NVIDIAGameWorks/kaolin'
-AUTHOR = ''
+AUTHOR = 'Nvidia'
```
Comment (krrish94) [inline setup.py]: Should double-check whether we write `Nvidia` or `NVIDIA`

Comment (Jean-Francois-Lafleche) [reply setup.py]: Based on here: https://nvidia.frontify.com/d/qZxUpjTCnt96/nvidia-logo-quickstart-guide, I believe you are correct. Good catch!

File: setup2.py (hunk 1 above)
Comment (Jean-Francois-Lafleche) [reply setup2.py]: It is, you are looking at a deleted file.

File: setup2.py (hunk 2 above)
Comment (Jean-Francois-Lafleche) [reply setup2.py]: setup2.py is deleted

File: setup2.py (hunk 3 above)
Comment (Jean-Francois-Lafleche) [reply setup2.py]: setup2.py is deleted

File: kaolin/cython/mcubes.cpp [hunk 5]
```diff
@@ -891,7 +877,7 @@ static const char *__pyx_f[] = {
   "type.pxd",
 };
 
-/* "../../.local/lib/python3.6/site-packages/Cython/Includes/numpy/__init__.pxd":776
+/* "../../../miniconda3/envs/temptemp37/lib/python3.7/site-packages/Cython-0.29.14-py3.7-linux-x86_64.egg/Cython/Includes/numpy/__init__.pxd":776
```
Comment (Caenorst) [inline kaolin/cython/mcubes.cpp]: I see that there is a lot of those diff in your PR, can you explain what is it doing ? (It seems commented, right ?)

**Review by Caenorst — COMMENTED** [review]
The description of the PR says that the test was:
`python setup.py build_ext --inplace; python setup.py install;`
Is `python setup.py build_ext --inplace` required to installed kaolin now ? If yes shouldn't it be described in the README ? What happens if you only do `python setup.py install` ?

File: kaolin/cython/mcubes.cpp (hunk 5 above)
Comment (Jean-Francois-Lafleche) [reply kaolin/cython/mcubes.cpp]: Hmm, these files are generated by Cython, so I didn't dig into them You're right, I'll go through and clean them up.

Comment (Caenorst) [reply kaolin/cython/mcubes.cpp]: If the files are generated by Cython, should we actually put them in the repository ? Shouldn't `setup.py` generate them for the user ?

**Comment by Jean-Francois-Lafleche** [comment]
> The description of the PR says that the test was:
> `python setup.py build_ext --inplace; python setup.py install;`
> Is `python setup.py build_ext --inplace` required to installed kaolin now ? If yes shouldn't it be described in the README ? What happens if you only do `python setup.py install` ?

`python setup.py install` will work by itself in that it will install kaolin to the user's current environment's site-packages. However, imports from the kaolin root dir will fail as described above. So while `python setup.py build_ext --inplace` is optional (as indicated in the README), there's no cost to doing this command first (the subsequent install does not have to rebuild the extensions) and it avoids the aforementioned import issue.

For development, the command to run is still simply `python setup.py develop`.

All that said, I'll update the description to note that I've tested doing the install step by itself as well.

File: kaolin/cython/mcubes.cpp (hunk 5 above)
Comment (Jean-Francois-Lafleche) [reply kaolin/cython/mcubes.cpp]: See the first point in the description.

**Comment by krrish94** [comment]
LGTM

File: kaolin/cython/mcubes.cpp (hunk 5 above)
Comment (Jean-Francois-Lafleche) [reply kaolin/cython/mcubes.cpp]: Comments (original source code) are now omitted from the generated C files.

**Pull request merged**